books/free-programming-books-subjects.md

```diff
@@ -397,6 +397,8 @@ Books that cover a specific programming language can be found in the [BY PROGRAM
 * [Foundations of Machine Learning, Second Edition](https://mitpress.ublish.com/ebook/foundations-of-machine-learning--2-preview/7093/Cover) - Mehryar Mohri, Afshin Rostamizadeh, Ameet Talwalkar
 * [Free and Open Machine Learning](https://nocomplexity.com/documents/fossml/) - Maikel Mardjan (HTML)
 * [Gaussian Processes for Machine Learning](https://www.gaussianprocess.org/gpml/) - Carl Edward Rasmussen, Christopher K.I. Williams
+* [Hands-On Machine Learning with Scikit-Learn and TensorFlow: Concepts, Tools, and Techniques to Build Intelligent Systems
+](https://shorturl.at/djtEI) - Geron Aurelien (PDF)
```

Comment:
We don't accept shortened links; we prefer original sources. Share the direct source for consideration.

Reply:
sure.